libnymea-core/zigbee/zigbeemanager.h

```diff
@@ -62,7 +62,8 @@ class ZigbeeManager : public QObject
         ZigbeeErrorNetworkOffline,
         ZigbeeErrorUnknownBackend,
         ZigbeeErrorNodeNotFound,
-        ZigbeeErrorForbidden
+        ZigbeeErrorForbidden,
+        ZigbeeErrorInvalidChannel,
```

Comment:
You don't seem to use this error code, you could make sure the given uint mask is a uint32, but since you say `channelMask = params.value("channelMask").toUInt() & ZigbeeChannelMask::ChannelConfigurationAllChannels` you always get a valid mask.

Reply:
Maybe check if the resulting channelmask is 0, since that might be possible and no channel for scanning is not allowed. Depends on the firmware how they handle a 0 channelmask. I always enable the primary light link channel is I get a 0 channel mask on the firmware side, but that might not always be the case.

Reply:
Very fair point. Done. Thanks